framework/packages/abstract-interface/src/migrate.rs

```diff
@@ -37,4 +43,105 @@ impl<T: CwEnv> Abstract<T> {
+            != contract_version(&self.account_factory)?.version
+        {
+            let migration_result = self
+                .account_factory
+                .upload_and_migrate_if_needed(&account_factory::MigrateMsg {})?;
+            if migration_result.is_some() {
+                has_migrated = true;
+            }
+        }
+        if ::module_factory::contract::CONTRACT_VERSION
+            != contract_version(&self.module_factory)?.version
+        {
+            let migration_result = self
+                .module_factory
+                .upload_and_migrate_if_needed(&module_factory::MigrateMsg {})?;
+            if migration_result.is_some() {
+                has_migrated = true;
+            }
+        }
+
+        if ::version_control::contract::CONTRACT_VERSION
+            != contract_version(&self.version_control)?.version
+        {
+            let migration_result = self
+                .version_control
+                .upload_and_migrate_if_needed(&version_control::MigrateMsg {})?;
+            if migration_result.is_some() {
+                has_migrated = true;
+            }
+        }
+
+        if ::ans_host::contract::CONTRACT_VERSION != contract_version(&self.ans_host)?.version {
+            let migration_result = self
+                .ans_host
+                .upload_and_migrate_if_needed(&ans_host::MigrateMsg {})?;
+            if migration_result.is_some() {
+                has_migrated = true;
+            }
+        }
+
+        let mut modules_to_register = Vec::with_capacity(2);
+
+        // We need to check the version in version control for the account contracts
+        let versions = self
+            .version_control
+            .modules(vec![
+                ModuleInfo::from_id_latest(MANAGER)?,
+                ModuleInfo::from_id_latest(PROXY)?,
+            ])?
+            .modules;
+
+        if ::manager::contract::CONTRACT_VERSION != versions[0].module.info.version.to_string()
+            && self.account.manager.upload_if_needed()?.is_some()
+        {
+            modules_to_register.push((
+                self.account.manager.as_instance(),
+                ::manager::contract::CONTRACT_VERSION.to_string(),
+            ));
+        }
+
+        if ::proxy::contract::CONTRACT_VERSION != versions[1].module.info.version.to_string()
+            && self.account.proxy.upload_if_needed()?.is_some()
+        {
+            modules_to_register.push((
+                self.account.proxy.as_instance(),
+                ::proxy::contract::CONTRACT_VERSION.to_string(),
+            ));
+        }
+
+        if !modules_to_register.is_empty() {
+            self.version_control
+                .register_account_mods(modules_to_register)?;
+            has_migrated = true
+        }
+
+        // TODO ibc client
+
+        self.version_control.approve_any_abstract_modules()?;
+
+        Ok(has_migrated)
+    }
+}
+
+fn contract_version<Chain: CwEnv, A: ContractInstance<Chain>>(
```

Comment:
Would be great to have this as a query on cw-orch!

Reply:
You think ? This is cw2 specific, so not sure it belongs in cw-orch. But could be included in the cw2 interface, dunno how re-usable that could be

Reply:
> You think ? This is cw2 specific, so not sure it belongs in cw-orch. But could be included in the cw2 interface, dunno how re-usable that could be

we can make it as sub-interface of WasmQuerier with `cw-plus` feature enabled, wdyt? And use like `chain.wasm_querier().cw2().contract_info(addr)?`